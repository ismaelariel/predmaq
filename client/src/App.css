body {
    margin: 0;
    padding: 0;
    width: 100%;
    height: 100%;
    position: absolute;
    font-family: Arial, Helvetica, sans-serif;
}
