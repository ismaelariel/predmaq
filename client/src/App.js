import React from "react";
import {BrowserRouter, Routes, Route} from "react-router-dom";

import Machine from "./MachinePage/Machine";

import "./App.css";

const App = () => {
    return(
        <BrowserRouter>
            <Routes>
                <Route path="/" element={<Machine/>}/>
            </Routes>
        </BrowserRouter>
    );
};

export default App;
