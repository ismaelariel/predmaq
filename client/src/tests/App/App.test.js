import React from "react";
import renderer from "react-test-renderer";

import App from "../../App";

const { MongoClient } = require('mongodb');

describe('App Component View', () => {
    let connection;
    let db;

    beforeAll(async () => {
        connection = await MongoClient.connect(globalThis.__MONGO_URI__, {
            useNewUrlParser: true,
            useUnifiedTopology: true,
        });
        db = await connection.db(globalThis.__MONGO_DB_NAME__);
    });

    afterAll(async () => {
        await connection.close();
    });

    it('test mongodb connection', async () => {
        const mcdb = db.collection('predmaq');

        const machine = {
            _id: '6443e224931711288579065c',
            UDI: 1,
            ProductID: 'M14860',
            Type: 'M',
            AirTemperature: 298.1,
            ProcessTemperature: 308.6,
            RotationalSpeed: 1551,
            Torque: 42.8,
            ToolWear: 0,
            Target: 0,
            FailureType: 'No Failure'
        };

        await mcdb.insertOne(machine);

        const insertedMachine = await mcdb.findOne({ _id: '6443e224931711288579065c' });
        expect(insertedMachine).toEqual(machine);
    });

    beforeEach(async () => {
        await db.collection('predmaq').deleteMany({});
    });

    it('render app root component', () => {
        const component = renderer.create(
            <App />,
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
