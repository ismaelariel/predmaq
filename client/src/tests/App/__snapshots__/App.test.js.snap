// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component View render app root and chart component 1`] = `
<div
  className="div_predict_container"
>
  <div
    className="div_navbar_container"
  >
    <div
      className="div_navbar_content div_navbar_content_app_name"
    >
      <h4>
        PredMaq
      </h4>
    </div>
    <div
      className="div_navbar_content div_navbar_content_datetime"
    >
      <span
        className="text_datatime_navbar"
      />
    </div>
  </div>
  <div
    className="div_body_container"
  >
    <div
      className="div_sidebar_left_content"
    >
      <div
        className="div_img_container"
      >
        <img
          alt="logo predmaq"
          id="id_img_predmaq"
          src="test-file-stub"
        />
      </div>
      <div
        className="div_action_container"
      >
        <div
          className="div_action_group_conatiner"
        >
          <input
            aria-label="cost-input"
            className="input_action_style action_style"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Procurar..."
            type="text"
            value=""
          />
        </div>
        <div
          className="div_action_group_conatiner"
        >
          <button
            className="btn_action_style action_style"
            onClick={[Function]}
          >
            Temperatura
          </button>
        </div>
        <div
          className="div_action_group_conatiner"
        >
          <button
            className="btn_action_style action_style"
            onClick={[Function]}
          >
            Rotação
          </button>
        </div>
        <div
          className="div_action_group_conatiner"
        >
          <button
            className="btn_action_style action_style"
            onClick={[Function]}
          >
            Desgaste
          </button>
        </div>
        <div
          className="div_action_group_conatiner"
        >
          <button
            className="btn_action_dinamic_style action_style"
            onClick={[Function]}
            style={
              {
                "backgroundColor": "#ff0000",
              }
            }
          >
            Pausada
          </button>
        </div>
      </div>
    </div>
    <div
      className="div_sidebar_right_content"
    >
      <div
        className="div_plot_conatiner"
      >
        <div
          className="div_plot_content"
        >
          <canvas
            height={100}
            role="img"
            width={100}
          />
        </div>
        <div
          className="div_plot_content"
        >
          <canvas
            height={100}
            role="img"
            width={100}
          />
        </div>
      </div>
      <div
        className="div_grid_conatiner"
      >
        <table
          className="table table table-striped"
        >
          <thead>
            <tr>
              <th>
                AirTemperature
              </th>
              <th>
                FailureType
              </th>
              <th>
                ProcessTemperature
              </th>
              <th>
                ProductID
              </th>
              <th>
                RotationalSpeed
              </th>
              <th>
                Target
              </th>
              <th>
                ToolWear
              </th>
              <th>
                Torque
              </th>
              <th>
                Type
              </th>
              <th>
                UDI
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`App Component View render app root component 1`] = `
<div
  className="div_predict_container"
>
  <div
    className="div_navbar_container"
  >
    <div
      className="div_navbar_content div_navbar_content_app_name"
    >
      <h4>
        PredMaq
      </h4>
    </div>
    <div
      className="div_navbar_content div_navbar_content_datetime"
    >
      <span
        className="text_datatime_navbar"
      />
    </div>
  </div>
  <div
    className="div_body_container"
  >
    <div
      className="div_sidebar_left_content"
    >
      <div
        className="div_img_container"
      >
        <img
          alt="logo predmaq"
          id="id_img_predmaq"
          src="test-file-stub"
        />
      </div>
      <div
        className="div_action_container"
      >
        <div
          className="div_action_group_conatiner"
        >
          <input
            aria-label="cost-input"
            className="input_action_style action_style"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Procurar..."
            type="text"
            value=""
          />
        </div>
        <div
          className="div_action_group_conatiner"
        >
          <button
            className="btn_action_style action_style"
            onClick={[Function]}
          >
            Temperatura
          </button>
        </div>
        <div
          className="div_action_group_conatiner"
        >
          <button
            className="btn_action_style action_style"
            onClick={[Function]}
          >
            Rotação
          </button>
        </div>
        <div
          className="div_action_group_conatiner"
        >
          <button
            className="btn_action_style action_style"
            onClick={[Function]}
          >
            Desgaste
          </button>
        </div>
        <div
          className="div_action_group_conatiner"
        >
          <button
            className="btn_action_dinamic_style action_style"
            onClick={[Function]}
            style={
              {
                "backgroundColor": "#ff0000",
              }
            }
          >
            Pausada
          </button>
        </div>
      </div>
    </div>
    <div
      className="div_sidebar_right_content"
    >
      <div
        className="div_plot_conatiner"
      >
        <div
          className="div_plot_content"
        >
          <canvas
            height={100}
            role="img"
            width={100}
          />
        </div>
        <div
          className="div_plot_content"
        >
          <canvas
            height={100}
            role="img"
            width={100}
          />
        </div>
      </div>
      <div
        className="div_grid_conatiner"
      >
        <table
          className="table table table-striped"
        >
          <thead>
            <tr>
              <th>
                AirTemperature
              </th>
              <th>
                FailureType
              </th>
              <th>
                ProcessTemperature
              </th>
              <th>
                ProductID
              </th>
              <th>
                RotationalSpeed
              </th>
              <th>
                Target
              </th>
              <th>
                ToolWear
              </th>
              <th>
                Torque
              </th>
              <th>
                Type
              </th>
              <th>
                UDI
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</div>
`;
